compiler/optimizer/VPHandlersCommon.cpp

```diff
@@ -69,8 +70,25 @@ TR::Node *constrainChildren(OMR::ValuePropagation *vp, TR::Node *node)
    TR::Node *myParent = vp->getCurrentParent();
    for (int32_t i = node->getNumChildren()-1; i >= 0; i--)
       {
+      TR::Node* child = node->getChild(i);
       vp->setCurrentParent(node);
-      vp->launchNode(node->getChild(i), node, i);
+      vp->launchNode(child, node, i);
+
+      // Replace PassThrough with its child
+      if (child->getOpCodeValue() == TR::PassThrough
+          && !node->getOpCode().isNullCheck())
```

Comment:
It's possible for `launchNode()` to replace the child, in which case `child` wouldn't be right anymore.

I thought `constrainChildren()` might be able to get away without having to test for null checks, but if not, perhaps this is better done inside `launchNode()`.

Should we use `performTransformation()` here?